cmd/tools/generate/generate.go

```diff
@@ -20,11 +29,55 @@ var systemdCmd = &cobra.Command{
 	Run:   doSystemd,
 }
 
+var dockerCmd = &cobra.Command{
+	Use:   "docker",
+	Short: "generate Harvest docker-compose.yml target for all pollers defined in config",
+	Run:   doDockerCompose,
+}
+
 func doSystemd(cmd *cobra.Command, _ []string) {
 	var config = cmd.Root().PersistentFlags().Lookup("config")
 	generateSystemd(config.Value.String())
 }
 
+func doDockerCompose(cmd *cobra.Command, _ []string) {
+	var config = cmd.Root().PersistentFlags().Lookup("config")
+	generateDockerCompose(config.Value.String())
+}
+
+func generateDockerCompose(path string) {
+	pollerTemplate := PollerTemplate{}
+	err := conf.LoadHarvestConfig(path)
+	if err != nil {
+		return
+	}
+	if conf.Config.Pollers == nil {
+		return
+	}
+	conf.IsDocker = true
+	for k, _ := range *conf.Config.Pollers {
```

Comment:
nicer to write these in the same order as listed in the yaml - straightforward with the `PollersOrdered` member of HarvestConfig

Reply:
done.